src/domain/products/details/variants/variant-editor.tsx

```diff
@@ -0,0 +1,362 @@
+import useMedusa from "../../../../hooks/use-medusa"
+import { countries as countryData } from "../../../../utils/countries"
+import { convertEmptyStringToNull } from "../../../../utils/convert-empty-string-to-null"
+import InfoTooltip from "../../../../components/molecules/info-tooltip"
+import CheckIcon from "../../../../components/fundamentals/icons/check-icon"
+import TrashIcon from "../../../../components/fundamentals/icons/trash-icon"
+import PlusIcon from "../../../../components/fundamentals/icons/plus-icon"
+import Select from "../../../../components/molecules/select"
+const numberFields = ["weight", "length", "width", "height"]
+
+const VariantEditor = ({
+  variant,
+  options,
+  onSubmit,
+  onDelete,
+  onCancel,
+  isCopy,
+}) => {
+  const countryOptions = countryData.map((c) => ({
+    label: c.name,
+    value: c.alpha2.toLowerCase(),
+  }))
+
+  const { store, isLoading } = useMedusa("store")
+  console.log(variant)
+  const [currencyOptions, setCurrencyOptions] = useState([])
+  const [prices, setPrices] = useState(variant.prices)
+  const [selectedCountry, setSelectedCountry] = useState(
+    variant.origin_country
+      ? countryOptions.find((cd) => cd.label === variant.origin_country)
+      : undefined
+  )
+
+  console.log("selected country", selectedCountry)
+  const { setValue, getValues, register, reset, watch, handleSubmit } = useForm(
+    variant
+  )
+
+  useEffect(() => {
+    reset({
+      ...variant,
+    })
+
+    variant.options.forEach((option, index) => {
+      register(`options.${index}.option_id`)
+      setValue(`options.${index}.option_id`, option.option_id)
+    })
+  }, [variant])
+
+  const getCurrencyOptions = () => {
+    return ((store && store.currencies) || [])
+      .map((v) => ({
+        value: v.code.toUpperCase(),
+        label: v.code.toUpperCase(),
+      }))
+      .filter(
+        (o) => !prices.find((p) => !p.edit && p.currency_code === o.value)
+      )
+  }
+
+  useEffect(() => {
+    setCurrencyOptions(getCurrencyOptions())
+  }, [store, isLoading, variant.prices])
+
+  const handleCurrencySelected = (index, currency) => {
+    const newPrices = [...prices]
+    newPrices[index] = {
+      ...newPrices[index],
+      currency_code: currency.toLowerCase(),
+    }
+
+    setPrices(newPrices)
+  }
+
+  const handlePriceChange = (index, value) => {
+    console.log(value)
+    const newPrices = [...prices]
+    newPrices[index] = {
+      ...newPrices[index],
+      amount: value,
+    }
+
+    setPrices(newPrices)
+  }
+
+  const removePrice = (index) => {
+    const newPrices = [...prices]
+    newPrices.splice(index, 1)
+    setPrices(newPrices)
+  }
+
+  const addPrice = () => {
+    const newPrices = [
+      ...prices,
+      {
+        edit: true,
+        region: "",
+        currency_code: currencyOptions[0].value,
+        amount: "",
+        sale_amount: "",
+      },
+    ]
+
+    setPrices(newPrices)
+  }
+
+  const handleSave = (data) => {
```

Comment:
Saving in the variant editor should trigger an update call to the backend instead of closing the modal

Reply:
And the same goes for when you've copied a variant and want to create it.